packages/medusa/src/services/cart.ts

```diff
@@ -1424,12 +1427,17 @@ class CartService extends TransactionBaseService {
     })
 
     if (itemsToRemove.length) {
-      const results = await promiseAll(
+      await promiseAll(
         itemsToRemove.map(async (item) => {
           return this.removeLineItem(cart.id, item.id)
         })
       )
-      cart.items = results.pop()?.items ?? []
+
+      const updatedCart = await this.retrieve(cart.id, {
+        relations: ["items.variant.product.profiles"],
+      })
+
+      cart.items = updatedCart.items ?? []
```

Comment:
q: This isn't really needed, is it?

Reply:
sorry, for some reason, it didn't add it to my approval comment.

Reply:
it is needed, since the items on the cart needs to be removed for later usage, otherwise, you need to refresh the cart anyway just outside, but here the idea of that method is to update the cart items ref to remove the removed items. I ll change that though to not re fetch but instead just delete them in place wdyt?

Reply:
what do you think of that https://github.com/medusajs/medusa/pull/5980/commits/d31ba4685e43176ce7c5bab9154777ce52f245b7

Reply:
wait, let me improve

Reply:
yeah, the ideal would be the consumer to get a fresh cart, not this method to return a fresh cart cause in most cases it will not included all the expected relations that the consumer expect to have. Therefore, I think those methods should in reality only focus on doing those operations and if needed the consumer refresh its cart. but here I didn't want to change the behaviour to mcuh since I v already changed some behaviours to cleanup ahah. But you are right